blockchain/contracts/Staking.sol

```diff
@@ -30,7 +31,7 @@ contract Staking is IERC721Receiver {
     // Allows only one NFT staked at a time per user
     modifier onlyOneStake() {
         require(
-            this.isStaking(msg.sender),
+            !isStaking(msg.sender),
```

Comment:
Good catch! Obviously not required to do for this project but this is an example where unit tests would have been useful :P